mm.c

```diff
@@ -169,13 +169,55 @@ static void *find_free_block(size_t words) {
 }
 
 /*
-	Assume that size in words given is <= the size of the block at input.
-	bp input is the block that you found already that is large enough.
-	The function reduces the size of the block if it is too large.
+	The function takes free block and changes status to taken.
+	The function reduces the size of the block (splits it) if size is too large.
 	The remaining size is either placed in free_list or left hanging if it is 0.
+	bp input is the block that you found already that is large enough.
+	Assume that size in words given is <= the size of the block at input.
 */
 static void alloc_free_block(void *bp, size_t words) {
+	size_t bp_size = GET_SIZE(bp);
+	size_t bp_tot_size = bp_size + HDR_FTR_SIZE;
+
+	size_t needed_size = words;
+	size_t needed_tot_size = words + HDR_FTR_SIZE;
+
+	size_t new_block_size;
+	size_t new_block_tot_size;
+
+	// the block created from extra free space
+	char **new_block;
+
+	// to allocate block, first remove it from free list
+	remove_block_from_free_list(bp);
+
+	// if size of block is larger than needed size, split the block
+	// handle new block by making it part of the free block ecosystem
+	if (bp_size > needed_size) {
```

Comment:
So instead return a block of size 2 words larger?

I guess that will be fine too since we are rounding up the bytes in malloc anyway.

Still, I will work on this function after coalesce is merged.